homeassistant/components/http/__init__.py

```diff
@@ -264,6 +277,24 @@ async def start_server(*_: Any) -> None:
         local_ip, host, server_port, ssl_certificate is not None
     )
 
+    @callback
+    def _async_check_ssl_issue(_: Event) -> None:
+        if (
+            (hass.config.external_url is None or hass.config.internal_url is None)
+            and ssl_certificate is not None
+            and not is_cloud_connection(hass)
```

Comment:
This will always be false as the event is fired locally.
`is_cloud_connection(hass)` will be only True if the function call originated from a cloud request.